core/whistle-1.0.0/src/wh_cache.erl

```diff
@@ -143,6 +143,15 @@ start_link(Name, ExpirePeriod, Props) ->
                                    )
     end.
 
+-spec build_bindings(wh_proplist()) -> wh_proplist().
+build_bindings([[]]) -> [{'conf', ['federate']}];
+build_bindings(BindingProps) ->
+     add_database_binding([{'conf', ['federate'|P]} || P <- BindingProps]).
```

Comment:
Personally, I'd do

```erlang
build_bindings(BindingProps) ->
    [database_binding() | document_bindings(BindingProps)].

document_bindings(BindingProps) ->
    [{'conf', ['federate'|P]} || P <- BindingProps].

database_binding() ->
    {'conf', ['federate', {'type', <<"database">>}]}.
```

`build_bindings/1` is then easily readable without having to know what those binding tuples look like.

Thoughts?